From 8e8be21a1e740947cedc8bee6bf2ab7bec516e45 Mon Sep 17 00:00:00 2001
From: Sylvain Potel <nxf82999@lsv05577.swis.nl-cdc01.nxp.com>
Date: Fri, 3 Mar 2023 13:17:15 +0100
Subject: [PATCH 1/1] Issue: WSW-25718 [Firecrest][15.4][SPI] increase
 SPI_DEFAULT_SMALL_PACKET_SIZE. [Description] To improve SPI data transfer
 increase default minimal packet size from 32 to 48 bytes. This is mandatory
 to have SPI driver align with this value as well. Update spifreq CLI command
 as now default mode supports up to 4MHz, High Speed until 10MHz and Full
 speed until 16.6MHz(=20MHz in ot-daemon input argument).

---
 src/posix/platform/daemon.cpp                             | 8 ++++----
 src/posix/platform/include/openthread/openthread-system.h | 2 +-
 2 files changed, 5 insertions(+), 5 deletions(-)

diff --git a/src/posix/platform/daemon.cpp b/src/posix/platform/daemon.cpp
index f4f665ea1..ab1705c5f 100644
--- a/src/posix/platform/daemon.cpp
+++ b/src/posix/platform/daemon.cpp
@@ -623,14 +623,14 @@ void ProcessGetSetSpiFrequency(void *aContext, uint8_t aArgsLength, char *aArgs[
         switch(speed)
         {
             case 1: //OT_RADIO_HIGH_BUS_SPEED
-                otCliOutputFormat("Get => RCP supports SPI Bus Speed until 6MHz\r\n");
+                otCliOutputFormat("Get => RCP supports SPI Bus Speed until 10MHz\r\n");
             break;
             case 2: //OT_RADIO_FULL_BUS_SPEED
-                otCliOutputFormat("Get => RCP supports SPI Bus Speed until 10MHz\r\n");
+                otCliOutputFormat("Get => RCP supports SPI Bus Speed > 10MHz\r\n");
             break;
             case 0: //OT_RADIO_DEFAULT_BUS_SPEED
             default:
-                otCliOutputFormat("Get => RCP supports SPI Bus Speed until 3MHz\r\n");
+                otCliOutputFormat("Get => RCP supports SPI Bus Speed until 4MHz\r\n");
             break;
         }
     }
@@ -638,7 +638,7 @@ void ProcessGetSetSpiFrequency(void *aContext, uint8_t aArgsLength, char *aArgs[
     {
         uint32_t speed = (uint32_t)atoi(aArgs[0]);
 
-        if( (speed >= 10000) && (speed <= 10000000) )
+        if( (speed >= 10000) && (speed <= 20000000) )
         {
             if( otThreadGetDeviceRole((otInstance*)aContext) == OT_DEVICE_ROLE_DISABLED )
             {
diff --git a/src/posix/platform/include/openthread/openthread-system.h b/src/posix/platform/include/openthread/openthread-system.h
index e8210703b..b34f953ae 100644
--- a/src/posix/platform/include/openthread/openthread-system.h
+++ b/src/posix/platform/include/openthread/openthread-system.h
@@ -62,7 +62,7 @@ enum
     OT_PLATFORM_CONFIG_SPI_DEFAULT_ALIGN_ALLOWANCE =
         16, ///< Default maximum number of 0xFF bytes to clip from start of MISO frame.
     OT_PLATFORM_CONFIG_SPI_DEFAULT_SMALL_PACKET_SIZE =
-        32,                                ///< Default smallest SPI packet size we can receive in a single transaction.
+        48,                                ///< Default smallest SPI packet size we can receive in a single transaction.
     OT_PLATFORM_CONFIG_MAX_RADIO_URLS = 2, ///< Max number of Radio URLs.
 };
 
